gusb/gusb-device.c

```diff
@@ -1395,15 +1396,24 @@ g_usb_device_get_hid_descriptor_for_interface(GUsbDevice *self, GUsbInterface *i
 					   G_USB_DEVICE_REQUEST_TYPE_STANDARD,
 					   G_USB_DEVICE_RECIPIENT_INTERFACE,
 					   LIBUSB_REQUEST_GET_DESCRIPTOR,
-					   LIBUSB_DT_REPORT << 8,
-					   g_usb_interface_get_number(intf),
+					   (LIBUSB_DT_REPORT << 8) |
+					       g_usb_interface_get_number(intf),
+					   g_usb_interface_get_index(intf),
 					   buf2,
 					   buf2sz,
 					   &actual_length,
 					   5000,
 					   NULL,
-					   error)) {
-		g_prefix_error(error, "failed to get HID report descriptor: ");
+					   &error_local)) {
+		if (g_error_matches(error_local,
+				    G_USB_DEVICE_ERROR,
+				    G_USB_DEVICE_ERROR_NOT_SUPPORTED)) {
+			g_debug("ignoring: %s", error_local->message);
```

Comment:
I kind of feel that it would be more correct to report an error than an empty buffer

Reply:
I initially thought so too; but I've yet to find a device that reports two HID descriptors and actually allows you to request the non-default one. I suspect we have a deeper bug somewhere, but interestingly `sudo lsusb -vvvv` also shows the same fault.